tests/structures/pseudo/test_expressions.py

```diff
@@ -180,6 +196,60 @@ def test_copy(self):
         assert id(original) != id(copy) and original == copy
 
 
+class TestNotUseableConstant:
+    def test_str(self):
+        assert str(NotUseableConstant(str(inf))) == "inf"
+        assert str(NotUseableConstant(str(-inf))) == "-inf"
+        assert str(NotUseableConstant(str(nan))) == "nan"
+
+    def test_requirements(self):
+        "Should not have requirements"
+        assert NotUseableConstant(str(inf)).requirements == []
+
+    def test_complexity(self):
+        "Should have a complexity of 1"
+        assert NotUseableConstant(str(inf)).complexity == 1
+
+    def test_equal(self):
+        "Should at least compare the id of the objects (inf == inf)"
+        assert NotUseableConstant(str(inf)) != NotUseableConstant(str(-inf))
+        assert NotUseableConstant(str(inf)) == NotUseableConstant(str(inf))
+
+    def test_repr(self):
+        assert repr(NotUseableConstant(str(inf))) == "inf type: not-usable-constant"
+        assert repr(NotUseableConstant(str(-inf))) == "-inf type: not-usable-constant"
+        assert repr(NotUseableConstant(str(nan))) == "nan type: not-usable-constant"
+
+    def test_substitute(self):
+        """Substitute shall have no effect when called directly on a variable."""
+        v = NotUseableConstant(str(inf))
+        v.substitute(v, NotUseableConstant(str(-inf)))
+        assert v == NotUseableConstant(str(inf))
+
+    def test_iter(self):
+        assert list(NotUseableConstant(str(inf))) == []
+        assert list(NotUseableConstant(str(inf))) == []
+        assert list(NotUseableConstant(str(inf))) == []
+
+    def test_copy(self):
+        original = NotUseableConstant(str(inf))
+        copy = original.copy()
+        assert id(original) != id(copy) and original == copy
+
+    def test_logic_converter_z3(self):
+        logic_converter: BaseConverter = Z3Converter()
+        instr1 = _get_condition_branch(Constant(inf, Float.double()))
+        instr2 = _get_condition_branch(NotUseableConstant(str(inf)))
+
+        # Not handled by 'Dead Path Elimination' yields OverflowError
+        with pytest.raises(OverflowError):
+            logic_converter.convert(instr1, define_expr=True)
+
+        # Covered by 'Dead Path Elimination' yields ValueError (will be skipped for z3 stuff)
+        with pytest.raises(ValueError):
+            logic_converter.convert(instr2, define_expr=True)
```

Comment:
This should be part of the z3 converter tests.

Reply:
Moved